base/complex.jl

```diff
@@ -420,8 +420,16 @@ sqrt(z::Complex) = sqrt(float(z))
 #     return Complex(abs(iz)/r/2, copysign(r,iz))
 # end
 
+function cis(v::Union{Float64,Float32})
+    res = FastMath.cis_fast(v) # Note: not yet defined at this point
+    return Math.nan_dom_err(res, v)
+end
```

Comment:
Note that MPFR defines `mpfr_sin_cos`, which we can use for the `BigFloat` version of this method.